### PR TITLE
MAINT: Py3K array_as_buffer and gentype_as_buffer

### PR DESCRIPTION
**Pull request by sethtroisi**
More Py3k work

### CODE REVIEW

File: numpy/core/src/multiarray/scalartypes.c.src [hunk 1]
```diff
@@ -2588,48 +2588,7 @@ gentype_getreadbuf(PyObject *self, Py_ssize_t segment, void **ptrptr)
...
-                "Non-character array cannot be interpreted "\
-                "as character buffer.");
-        return -1;
-    }
-}
-#endif /* !defined(NPY_PY3K) */
-
 static PyBufferProcs gentype_as_buffer = {
-#if !defined(NPY_PY3K)
-    gentype_getreadbuf,                          /* bf_getreadbuffer*/
```
Comment (sethtroisi) [inline numpy/core/src/multiarray/scalartypes.c.src]: gentype_getreadbuf is still used in gentype_byteswap

File: doc/Py3K.rst.txt [hunk 2]
```diff
@@ -394,28 +394,6 @@ There are a couple of places that need further attention:
   In some cases, this returns a buffer object on Python 2. On Python 3,
   there is no stand-alone buffer object, so we return a byte array instead.
 
-The Py2/Py3 compatible PyBufferMethods definition looks like::
```
Comment (eric-wieser) [inline doc/Py3K.rst.txt]: We decided not to touch this document, right @charris?

File: numpy/core/src/multiarray/scalartypes.c.src (hunk 1 above)
Comment (eric-wieser) [reply numpy/core/src/multiarray/scalartypes.c.src]: Can you maybe add a comment to gentype_getreadbuf like:
```
/* This function matches the Python 2.7 PyBufferProcs.bf_getreadbuffer interface, but no longer needs to.
 * In future we could consider rewriting callers to use `gentype_getbuffer`, or inline the function body at the caller.
 */
```

File: doc/Py3K.rst.txt (hunk 2 above)
Comment (charris) [reply doc/Py3K.rst.txt]: Right. Maybe we should have a place to put "history". The original intent of the document was to aid others in making the transition at the C level. NumPy was one of the first to support Python 3.

Comment (mattip) [reply doc/Py3K.rst.txt]: `doc/Py3k.rst.txt` is not linked into the html documentation as far as I can tell. So we could move it to an `archive` directory

File: numpy/core/src/multiarray/scalartypes.c.src (hunk 1 above)
Comment (sethtroisi) [reply numpy/core/src/multiarray/scalartypes.c.src]: Added at function definition, thanks for the well worded comment.

File: doc/Py3K.rst.txt (hunk 2 above)
Comment (sethtroisi) [reply doc/Py3K.rst.txt]: reverted.

**Review by eric-wieser — APPROVED** [review]

**Comment by mattip** [comment]
Thanks @sethtroisi

**Pull request merged**